### PR TITLE
Prac 09 feedback

### PR DESCRIPTION
**Pull request by RemoWeiersmuller**
Hi @ctown0711, 

please review my PR from prac_09. Focus on naming, SRP, DRY. 
I appreciate your feedback.

Cheers,

Remo

### CODE REVIEW

File: prac_09/band.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class Band:
+    def __init__(self, name: str):
+        """Construct the class."""
+        self.name = name
+        self.musicians = []
+
+    def __str__(self):
+        """Return a string of the object."""
+        musicians_string = ", ".join(str(musician) for musician in self.musicians)
+        return f"{self.name} ({musicians_string})"
```
Comment (ctown0711) [inline prac_09/band.py:17]: Use of the variable makes it neater, nice

File: prac_09/taxi.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+class Taxi(Car):
+    """Specialised version of a Car that includes fare costs."""
+    price_per_km = 1.23
+
+    def __init__(self, name, fuel):
+        """Initialise a Taxi instance, based on parent class Car."""
+        super().__init__(name, fuel)
+
+        self.current_fare_distance = 0
```
Comment (ctown0711) [inline prac_09/taxi.py:16]: Any reason for the blank line?

File: prac_09/unreliable_car.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        super().__init__(**kwargs)
+        self.reliability = reliability
+
+    def drive(self, distance):
+        """Drive the car, only if random number is less than reliability."""
+        if randint(0, 100) >= self.reliability:
+            distance = 0
+
+        distance_driven = super().drive(distance)
+        return distance_driven
```
Comment (ctown0711) [inline prac_09/unreliable_car.py:20]: Could you just return super().drive(distance) ?

File: prac_09/taxi_simulator.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        if menu_choice == 'C':
+            print('Taxi available:')
+            display_taxis(taxis)
+            taxi_choice = input("Choose taxi: ")
+            try:
+                current_taxi = taxis[int(taxi_choice)]
+            except IndexError:
+                print('Invalid taxi choice')
+            except ValueError:
+                print('Choice has to be number')
```
Comment (ctown0711) [inline prac_09/taxi_simulator.py:31]: Nothing in the sample output for this but nice job :)

File: prac_09/taxi_simulator.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        menu_choice = input('>>>').upper()
+    print(f"Bill to date: ${bill_to_date:.2f}")
+    print('Taxi are now:')
+    display_taxis(taxis)
+
+
+def display_taxis(taxis):
+    """Display all taxis with a number to choose."""
+    for i, taxi in enumerate(taxis):
+        print(f"{i} - {taxi}")
```
Comment (ctown0711) [inline prac_09/taxi_simulator.py:54]: Good use of enumerate

File: prac_09/taxi_simulator.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    while value == '':
+        try:
+            value = float(input(prompt))
+            if value <= 0:
+                print('Number must be > 0')
+                value = ''
+        except ValueError:
+            print('Invalid input; enter a valid number.')
+            value = ''
+    return value
```
Comment (ctown0711) [inline prac_09/taxi_simulator.py:69]: Also a nice addition

**Review by ctown0711 — COMMENTED** [review]
Looks basically perfect to me, maybe another set of eyes would find something that I didn't, great code.

File: prac_09/unreliable_car.py (hunk 3 above)
Comment (RemoWeiersmuller) [reply prac_09/unreliable_car.py:20]: good hint 👍

**Comment by RemoWeiersmuller** [comment]
Thank you for the nice review 👍

**Pull request merged**